core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -159,6 +159,15 @@ private[spark] class TaskSchedulerImpl(
     this.synchronized {
       val manager = createTaskSetManager(taskSet, maxTaskFailures)
       activeTaskSets(taskSet.id) = manager
+      val attemptsByStage = activeTaskSets.values.map(_.taskSet).groupBy(_.stageId)
```

Comment:
its intentionally done this way.  we shouldn't even have two task sets for the same stage.  (At least, I don't think it should.)

Also, `taskSet.id` includes the stage & attempt, so all the attempts do have distinct keys in `activeTasksSets` -- this is verified by the unit test and I've also seen this in real tests I've run.